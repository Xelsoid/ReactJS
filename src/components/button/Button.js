import React from 'react';
import PropTypes from 'prop-types';

import './button.scss';

const Button = ({btnClass, disabled, id, title, callback}) => {
  return (
    <button
      className={`btn ${btnClass}`}
      id={id}
      disabled={disabled}
      onClick={callback}
      type="button"
    >
      {title}
    </button>
  );
};

Button.defaultProps = {
  btnClass: 'btn--primary',
  disabled: true,
  title: 'disabled',
  callback: null,
  id: '',
};

Button.propTypes = {
  btnClass: PropTypes.string,
  disabled: PropTypes.bool,
  title: PropTypes.string,
  callback: PropTypes.func,
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

export default Button;
