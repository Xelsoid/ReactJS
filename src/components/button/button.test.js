import React from 'react';
import Button from './Button';

describe('Button', () => {
  it('renders correctly', () => {
    const component = shallow(
      <Button
        id='default'
        title='Default'
        disabled={false}
        btnClass='btn--primary'
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('clicked', () => {
    const component = shallow(
      <Button
        id='default'
        title='Default'
        disabled={false}
        btnClass='btn--primary'
      />
    );
    component.simulate('click');
    expect(component.text()).toMatch('Default clicked')
  });
});
