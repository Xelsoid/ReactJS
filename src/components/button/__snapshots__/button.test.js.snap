// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="btn btn--primary"
  disabled={false}
  id="default"
  onClick={null}
  type="button"
>
  Default
</button>
`;
